opm/flowdiagnostics/TracerTofSolver.cpp

```diff
@@ -128,38 +212,79 @@ namespace FlowDiagnostics
 
 
 
-    void TracerTofSolver::solveSingleCell(const int cell)
+    void TracerTofSolver::computeLocalOrdering(const CellSet& startset)
     {
-        // Compute downwind fluxes.
-        double downwind_flux = 0.0;
-        for (const auto& conn : g_.cellNeighbourhood(cell)) {
-            downwind_flux += conn.weight;
+        // Extract start cells.
+        std::vector<int> startcells(startset.begin(), startset.end());
+
+        // Compute reverse topological ordering.
+        const size_t num_cells = pv_.size();
+        assert(g_.startPointers().size() == num_cells + 1);
+        struct ResultDeleter { void operator()(TarjanSCCResult* x) { destroy_tarjan_sccresult(x); } };
+        std::unique_ptr<TarjanSCCResult, ResultDeleter> result;
+        {
+            struct WorkspaceDeleter { void operator()(TarjanWorkSpace* x) { destroy_tarjan_workspace(x); } };
+            std::unique_ptr<TarjanWorkSpace, WorkspaceDeleter> ws(create_tarjan_workspace(num_cells));
```

Comment:
There's nothing **wrong** with this, per se, but the `TarjanWorkSpace` isn't entirely trivial to construct either.  We exposed this type to the API level so that there would be an opportunity to reuse it for multiple sets of start points.